src/main/java/at/hannibal2/skyhanni/features/foraging/CompactSweepDetails.kt

```diff
@@ -0,0 +1,202 @@
+    private val patternGroup = RepoPattern.group("foraging.sweep-details")
+
+    /**
+     * REGEX-TEST: §6Sweep Details: §r§2442∮ Sweep
+     * REGEX-TEST: §6Sweep Details: §r§2442.76∮ Sweep
+     * REGEX-TEST: §6Sweep Details: §r§2451.65∮ Sweep
+     * REGEX-TEST: §6Sweep Details: §r§2430.65∮ Sweep
+     * REGEX-TEST: §6Sweep Details: §r§234,442.2∮ Sweep
+     */
+    @Suppress("MaxLineLength")
+    private val sweepDetailsPattern by patternGroup.pattern(
+        "header",
+        "(?:§.)+Sweep Details: (?<sweep>(?:§.)+(?<sweepAmt>[\\d,.]+)).? Sweep",
+    )
+
+    /**
+     * REGEX-TEST:   §r§7Fig Tree Toughness: §r§63.5 §r§a18.13 Logs
+     * REGEX-TEST:   §r§7Fig Tree Toughness: §r§63.5 §r§a18.19 Logs
+     * REGEX-TEST:   §r§7Fig Tree Toughness: §r§63.5 §r§818.19 Logs
+     * REGEX-TEST:   §r§7Fig Tree Toughness: §r§63.5 §r§818.04 Logs
+     * REGEX-TEST:   §r§7Fig Tree Toughness: §r§63.5 §r§818 Logs
+     * REGEX-TEST:   §r§7Dark Oak Tree Toughness: §r§60 §r§a35 Logs
+     */
+    @Suppress("MaxLineLength")
+    private val sweepToughnessLogsPattern by patternGroup.pattern(
+        "toughness-and-logs",
+        "\\s+(?:§.)+(?<treeType>[\\S ]+) Tree Toughness: (?<toughnessDisplay>§r§6(?<toughnessAmount>[\\d,.]+)) (?<logsDisplay>(?:§.)+(?<isItGreen>§.)(?<logsAmount>[\\d,.]+)) Logs",
+    )
+
+    /**
+     * REGEX-TEST:   §r§7Axe throw: §r§c-50% Sweep §r§a9.02 Logs
+     * REGEX-TEST:   §r§7Axe throw: §r§c-50% Sweep §r§89.02 Logs
+     * REGEX-TEST:   §r§7Wrong Style: §r§c-50% Sweep §r§a9.1 Logs §r§cCut the trunk first!!
+     * REGEX-TEST:   §r§7Wrong Style: §r§c-50% Sweep §r§a4.51 Logs §r§cCut the trunk first!!
+     * REGEX-TEST:   §r§7Wrong Style: §c-50% Sweep §a2.38 Logs §cCut branches and trunk first!!
+     * REGEX-TEST:   §r§7Wrong Style: §r§c-50% Sweep §r§a2.38 Logs §r§cCut branches and trunk first!!
+     */
+    @Suppress("MaxLineLength")
+    private val penaltyPattern by patternGroup.pattern(
+        "penalty",
+        "\\s+(?:§.)+(?<penaltyReason>[\\S ]+): (?<penaltyDisplay>(?:§.)+-(?<penaltyPercent>[\\d,.]+)%) Sweep (?<logsDisplay>(?:§.)?(?<isItGreen>§.)(?<logsAmount>[\\d,.]+)) Logs(?: (?<proTip>(?:§.)+[\\S ]+))?",
+    )
+
+    private var sweepDetailsVariablesDirty = false
+    private var isInsideSweepDetails = false
+    private var isFinalCalculation = false
+
+    private var addedInitialLogs = false
+    private var logCountDisplay = ""
+    private var logs = -1.0
+    private var proTip = ""
+    private val sweepDetailsChatBreakdown = mutableListOf<String>()
+    private var sweepDisplay = ""
+    private val sweepPenaltyHoverHistory = mutableListOf<String>()
+    private var sweep = -1.0
+    private var toughness = -1.0
+    private var toughnessDisplay = ""
+    private var treeType = ""
+
+    @HandleEvent
+    fun onChat(event: SkyHanniChatEvent) {
+        if (!isInIsland() || !config.compactSweepDetails) return
+        event.blockAndReadDetails()
```

Comment:
i dont see the benefit of this function.

Reply:
moved back into event body